source/SAMRAI/geom/CartesianEdgeDoubleConservativeLinearRefine.C

```diff
@@ -237,15 +375,276 @@ CartesianEdgeDoubleConservativeLinearRefine::refine(
+                      });
+
+                  pdat::parallel_for_all_x(
+                      slope_box, [=] SAMRAI_HOST_DEVICE(int i, int j, int k) {
+                         const double coef2i = 0.5 * (diff0(i - 1, j, k) + diff0(i, j, k));
+                         const double boundi = 2.0 * MIN(fabs(diff0(i - 1, j, k)), fabs(diff0(i, j, k)));
+
+                         if (diff0(i, j, k) * diff0(i - 1, j, k) > 0.0 && cdx0 != 0) {
+                            slope0(i, j, k) = COPYSIGN(MIN(fabs(coef2i), boundi), coef2i) / cdx0;
+                         } else {
+                            slope0(i, j, k) = 0.0;
+                         }
+
+                         const double coef2j = 0.5 * (diff1(i, j + 1, k) + diff1(i, j, k));
+                         const double boundj = 2.0 * MIN(fabs(diff1(i, j + 1, k)), fabs(diff1(i, j, k)));
+
+                         if (diff1(i, j, k) * diff1(i, j + 1, k) > 0.0 & cdx1 != 0) {
+                            slope1(i, j, k) = COPYSIGN(MIN(fabs(coef2j), boundj), coef2j) / cdx1;
+                         } else {
+                            slope1(i, j, k) = 0.0;
+                         }
+
+                         const double coef2k = 0.5 * (diff2(i, j, k - 1) + diff2(i, j, k));
+                         const double boundk = 2.0 * MIN(fabs(diff2(i, j, k - 1)), fabs(diff2(i, j, k)));
+
+                         if (diff2(i, j, k) * diff2(i, j, k - 1) > 0.0 && cdx2 != 0) {
+                            slope2(i, j, k) = COPYSIGN(MIN(fabs(coef2k), boundk), coef2k) / cdx2;
+                         } else {
+                            slope2(i, j, k) = 0.0;
+                         }
+                      
+                      });
+
+                  pdat::parallel_for_all_x(fine_box_plus,
+                      [=] SAMRAI_HOST_DEVICE(int i, int j, int k) {
+                         const int ic0 = (i < 0) ? (i + 1) / r0 - 1 : i / r0;
+                         const int ic1 = (j < 0) ? (j + 1) / r1 - 1 : j / r1;
+                         const int ic2 = (k < 0) ? (k + 1) / r2 - 1 : k / r2;
+
+                         const int ir0 = i - ic0 * r0;
+                         const int ir1 = j - ic1 * r1;
+                         const int ir2 = k - ic2 * r2;
+                         double deltax0, deltax1, deltax2;
+
+                         deltax0 = static_cast<double>(ir0) * fdx0;
+                         deltax1 = (static_cast<double>(ir1) + 0.5) * fdx1 - cdx1 * 0.5;
+                         deltax2 = static_cast<double>(ir2) * fdx2;
+
+                         double fine_tmp = coarse_array(ic0, ic1, ic2) +
+                                           slope0(ic0, ic1, ic2) * deltax0 +
+                                           slope1(ic0, ic1, ic2) * deltax1 +
+                                           slope2(ic0, ic1, ic2) * deltax2;
+                         fine_array(i, j, k) = fine_tmp;
+                        //  fprintf(stdout,
```

Comment:
Similar to comment in another file, is fine_tmp needed?

Reply:
Nope, this was a carry-over from a debugging session. So, you're right this is totally superfluous ;) I had it in mind to remove them at some point in a project wide clean-up. I'll take of them right away. Thanks @nselliott